apps/rule-manager/app/db/DbRuleDraft.scala

```diff
@@ -270,6 +270,44 @@ object DbRuleDraft extends SQLSyntaxSupport[DbRuleDraft] {
       case Left(result) => Left(result)
     }
   }
+  def batchUpdateFromFormRule(formRule: UpdateRuleForm, ids: List[Int], user: String)(implicit
+      session: DBSession = autoSession
+  ): Try[List[DbRuleDraft]] = {
+    try {
+      val updatedRules = ids.flatMap { id =>
+        val existingRule = find(id)
+          .toRight(
+            new Exception(s"Error updating rule with id ${id}: could not read existing rule")
+          )
+          .toOption
+
+        existingRule.map { rule =>
+          val mergedRule = rule.copy(
+            category = formRule.category.orElse(rule.category),
+            tags = formRule.tags.orElse(rule.tags)
+          )
+          val updatedRows = withSQL {
+            update(DbRuleDraft)
+              .set(
+                column.category -> mergedRule.category,
+                column.tags -> mergedRule.tags,
+                column.updatedBy -> user,
+                column.revisionId -> sqls"${column.revisionId} + 1"
+              )
+              .where
+              .eq(column.id, id)
```

Comment:
There's one drawback to this approach – we're making a database call to read and write for each rule we're updating. For larger numbers of rows, this will be very slow.

We should be able to write an update clause that uses an `in` condition to do this, something like:

```suggestion
              .in(column.id, listOfIdsToUpdate)
```

Then, we should be able to lose the `.map` in favour of a single DB call. I think the update call will fail if a rule does not exist, and that should be enough error handling for batch calls – I wouldn't worry about reading every rule to make sure it's there before we update. (We can always write a test to confirm, and if the error is very obscure we can consider making it more readable as we handle it.)

Full disclosure, although I'm confident this approach will work, I've never done this before with ScalikeJDBC, and it might be you've tried this and found problems with this approach – v. happy to pair!